IdentityCore/src/cache/token/MSIDCredentialCacheItem.m

```diff
@@ -314,6 +321,21 @@ - (BOOL)matchesWithRealm:(nullable NSString *)realm
         return YES;
     }
 
+    if ([NSString msidIsStringNilOrBlank:self.requestedClaims] && ![NSString msidIsStringNilOrBlank:requestedClaims])
```

Comment:
I'd like an extra set of eyes on this, as I think there may be a better/cleaner way to do this that I'm overlooking.

Here's a truth table of what I'm trying to do:

|     | A | B | nil | "" |
|-----|---|---|-----|----|
| A   | X |   |     |    |
| B   |   | X |     |    |
| nil |   |   | X   | X  |
| ""  |   |   | X   | X  |

I was having major problems doing string comparison when one side had a value and the other was a nil using IsEqual or IsEqualToString. Is there a string comparison function that would be better here?

Reply:
Would this work? 
result &= (!self.requestedClaims && !requestedClaims) || [self.requestedClaims isEqualToString:requestedClaims];

Reply:
Can the 2 conditions be separated out? Makes it less confusing imo : 
`if ([NSString msidIsStringNilOrBlank:self.requestedClaims] && [NSString msidIsStringNilOrBlank:requestedClaims]) {`
`           return YES;`
`}`

`if([self.requestedClaims isEqualToString:requestedClaims]) {`
 `          return YES;`
`}`
`return NO;`

or combine it to : 
`([NSString msidIsStringNilOrBlank:self.requestedClaims] && [NSString msidIsStringNilOrBlank:requestedClaims]) || ([self.requestedClaims isEqualToString:requestedClaims])`

Reply:
Thank you both very much for the help. Fixed in latest commit.